shared/actions/chat2/index.js

```diff
@@ -1247,99 +1080,234 @@ const messageSend = (action: Chat2Gen.MessageSendPayload, state: TypedState) =>
       tlfName,
       tlfPublic: false,
     }),
-    ...exitSearch,
   ])
 }
 
-// Start a conversation, or select an existing one
-const startConversation = (action: Chat2Gen.StartConversationPayload, state: TypedState) => {
-  const {participants, tlf, fromAReset} = action.payload
-  const you = state.config.username || ''
+const previewConversationAfterFindExisting = (
+  _fromPreviewConversation,
+  action: Chat2Gen.PreviewConversationPayload | Chat2Gen.SetPendingConversationUsersPayload
+) => {
+  // TODO make a sequentially that uses an object map and not all this array nonsense
+  if (!_fromPreviewConversation || _fromPreviewConversation.length !== 4) {
+    return
+  }
+  const results: ?RPCChatTypes.FindConversationsLocalRes = _fromPreviewConversation[1]
+  const users: Array<string> = _fromPreviewConversation[2]
 
-  let users: Array<string> = []
-  let conversationIDKey
+  let existingConversationIDKey
```

Comment:
maybe this should bail on creating the new inbox row if `existingConversationIDKey` matches the currently selected conversation? Right now I can be chatting with someone, open their tracker popup, click chat, and it'll insert a new row in the inbox for them even they're already selected. This could just do the navigation in that case

Reply:
this is on purpose. often the person isn't in the inbox list viewable so this solves that ux problem